packages/repo-tools/src/commands/api-reports/sql-reports/runSqlExtraction.ts

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs, { readJson } from 'fs-extra';
+import { relative as relativePath } from 'path';
+import { paths as cliPaths } from '../../../lib/paths';
+import { getPortPromise } from 'portfinder';
+import { diff as justDiff } from 'just-diff';
+import { SchemaInfo } from './types';
+import { getPgSchemaInfo } from './getPgSchemaInfo';
+import { generateSqlReport } from './generateSqlReport';
+import type { Knex } from 'knex';
+import { logApiReportInstructions } from '../api-extractor';
+
+interface SqlExtractionOptions {
+  packageDirs: string[];
+  isLocalBuild: boolean;
+}
+
+export async function runSqlExtraction(options: SqlExtractionOptions) {
+  const { default: Knex } = await import('knex');
+  const { default: EmbeddedPostgres } = await import('embedded-postgres').catch(
+    error => {
+      throw new Error(
+        `Failed to load peer dependency 'embedded-postgres' for generating SQL reports. ` +
+          `It must be installed as an explicit dependency in your project. Caused by; ${error}`,
+      );
+    },
+  );
+
+  const port = await getPortPromise();
+
+  const basePgOpts = {
+    host: 'localhost',
+    user: 'postgres',
+    password: 'password',
+  };
+  const pg = new EmbeddedPostgres({
+    databaseDir: './data/db',
+    ...basePgOpts,
+    port,
+    persistent: false,
+    onError(_messageOrError) {
+      // console.error('EmbeddedPostgres error:', messageOrError);
```

Comment:
Work in progress?

Reply:
Needs to be explicit no-op, but I'll clean these up